File: tribal-v2/tests/fixtures/sample_module/sample.go
// Tribal v2 ast_extract test fixture: Go.
package sample

import (
	"encoding/json"
	"fmt"
	"os"
)

type User struct {
	ID   string
	Name string
}

type UserService struct {
	dbPath string
}

func NewUserService(p string) *UserService {
	return &UserService{dbPath: p}
}

func (s *UserService) GetUser(uid string) (*User, error) {
	data, err := os.ReadFile(s.dbPath)
	if err != nil {
		return nil, err
	}
	var users map[string]*User
	if err := json.Unmarshal(data, &users); err != nil {
		return nil, err
	}
	return users[uid], nil
}

func Main() {
	svc := NewUserService("/tmp/db.json")
	u, _ := svc.GetUser("alice")
	fmt.Println(u.Name)
}

File: tribal-v2/tests/fixtures/sample_module/sample.py
"""Tribal v2 ast_extract test fixture: Python."""
import os
import json as _json
from pathlib import Path


class UserService:
    def get_user(self, uid: str) -> dict:
        return {"id": uid, "loaded_from": str(self._db_path())}

    def _db_path(self) -> Path:
        return Path(os.environ.get("DB", "/tmp/db.json"))


def main() -> None:
    svc = UserService()
    data = svc.get_user("alice")
    print(_json.dumps(data))
